### PR TITLE
Add code for multiproject GCE Client creation

### PR DESCRIPTION
**Pull request by panslava**
- Still uses default gce.conf to get most of the info
- If project info not specified will use default config values, which corresponds to cluster-project
- Overrides TokenURL and TokenBody and Network info if project is specified

/assign @gauravkghildiyal

### CODE REVIEW

**Review by gauravkghildiyal — COMMENTED** [review]
Thanks @panslava! Great unit tests.

File: pkg/multiproject/projectcrd/projectcrd.go [hunk 1]
```diff
@@ -19,6 +19,22 @@ func (p *Project) ProjectName() string {
 	return p.ObjectMeta.Labels[flags.F.MultiProjectCRDProjectNameLabel]
 }
 
+func (p *Project) ProjectID() string {
```
[the remark this answers is not in the document]
Comment (panslava) [reply pkg/multiproject/projectcrd/projectcrd.go]: Removed this, not needed for now

File: pkg/multiproject/gce/gce.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+// NewGCEForProject returns a new GCE client for the given project.
+// If the projectCRD is nil, it returns the default cloud, associated with the cluster-project.
+func NewGCEForProject(defaultConfigFile cloudgce.ConfigFile, projectCRD *projectcrd.Project, logger klog.Logger) (*cloudgce.Cloud, error) {
+	configFile, err := generateConfigFileForProject(defaultConfigFile, projectCRD)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate config file for project %s: %v", projectCRD.ProjectName(), err)
+	}
+
+	// convert file struct to bytes
+	configBytes, err := json.Marshal(configFile)
```
[the remark this answers is not in the document]
Comment (panslava) [reply pkg/multiproject/gce/gce.go]: Good catch

I changed the code significantly. Now it treats the file as ini file, and, I also updated unit tests to double check it, looks correct

File: pkg/multiproject/gce/gce.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	if err != nil {
+		klog.Fatalf("%v", err)
+	}
+	defer config.Close()
+
+	allConfig, err := io.ReadAll(config)
+	if err != nil {
+		klog.Fatalf("Error while reading config (%q): %v", flags.F.ConfigFilePath, err)
+	}
+	logger.V(4).Info("Cloudprovider config file", "config", string(allConfig))
```
[the remark this answers is not in the document]
Comment (panslava) [reply pkg/multiproject/gce/gce.go]: I moved a part of it into glbc/app, so we reuse some code, and it just makes it clearer :)

File: pkg/multiproject/gce/gce_test.go [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+
+				if expectedJSON != resultJSON {
+					t.Errorf("Expected result: %s, got: %s", expectedJSON, resultJSON)
+				}
+			}
+		})
+	}
+}
+
+func TestGenerateConfigForClusterSlice(t *testing.T) {
```
Comment (gauravkghildiyal) [inline pkg/multiproject/gce/gce_test.go:126]: Should we add a test-case to ensure other random fields which exist in a valid gce.conf are not modified and stay the same?

**Review by gauravkghildiyal — COMMENTED** [review]
Just one question, but besides that this looks ready.

File: pkg/multiproject/gce/gce_test.go (hunk 4 above)
Comment (panslava) [reply pkg/multiproject/gce/gce_test.go:126]: Good idea, added

**Comment by gauravkghildiyal** [comment]
Thanks!

/lgtm

**Pull request merged**